src/Arcus.Testing.Tests.Integration/Integration/DataFactory/RunDataFlowTests.cs

```diff
@@ -350,18 +350,22 @@ public static async Task<bool> IsDebugSessionActiveAsync(ResourceIdentifier reso
         /// </summary>
         public async ValueTask DisposeAsync()
         {
-            await using var disposables = new DisposableCollection(NullLogger.Instance);
-
-            if (Value != null)
+            await using (var disposables = new DisposableCollection(NullLogger.Instance))
             {
                 disposables.Add(AsyncDisposable.Create(async () =>
                 {
+                    // Expected to dispose the debug session multiple times to verify redundancy.
+                    await Value.DisposeAsync();
```

Comment:
Do you mean that you call `Dispose` multiple times to verify that no exception is thrown after the 2nd call to Dispose ?